spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/adapter/DelegatingInvocableHandler.java

```diff
@@ -125,22 +125,32 @@ protected InvocableHandlerMethod getHandlerForPayload(Class<? extends Object> pa
 	}
 
 	private void setupReplyTo(InvocableHandlerMethod handler) {
+		String replyTo = null;
 		Method method = handler.getMethod();
 		if (method != null) {
 			SendTo ann = AnnotationUtils.getAnnotation(method, SendTo.class);
-			if (ann != null) {
-				String[] destinations = ann.value();
-				if (destinations.length > 1) {
-					throw new IllegalStateException("Invalid @" + SendTo.class.getSimpleName() + " annotation on '"
-							+ method + "' one destination must be set (got " + Arrays.toString(destinations) + ")");
-				}
-				Address replyTo = destinations.length == 1 ? new Address(resolve(destinations[0])) : null;
-				if (replyTo != null) {
-					this.defaultReplyToForHandler.put(handler, replyTo);
-				}
-			}
+			replyTo = extractSendTo(method.toString(), ann);
+		}
+		if (replyTo == null) {
+			SendTo ann = AnnotationUtils.getAnnotation(this.bean.getClass(), SendTo.class);
```

Comment:
`@SendTo` is only method level annotation:
```
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface SendTo {
```
Or have you raised an SF JIRA on the matter already ?